### ISSUE TITLE
Weird behaviour with restore/save geomtery

### ISSUE THREAD

**Issue by steffenroeber** [githubuser0xFFFF/Qt-Advanced-Docking-System#470]
Hi,
I found some curious behaviour when using something like: (just pseudo example)
1. add a new dockwidget 
2. normal qt event loop processing...
3. save settings (saveGeometry, saveState, m_dockManager->saveState())
4. remove the dockwidget
5. add another dockwidget 
6. restore settings

Alls steps work properly if after step 4 normal qt event loop processing is done.
But, if step 3->6 are called without returning to top even event loop, then the widget to be removed is destroyed in step 4 is destroyed AFTER the new widget in step 5 is created. This mieght add curious effects in complex applicatiosn (e.g. in my for example).
After some debugging I found the reason for that:
In your lib basically all widgets use deleteLater logic instead of simple delete. The problems is, that in qt event loop these deferred delete events are ONlY processed in root event processing. Simply calling qApp->processEvents after step 4 is not sufficient, because the deleteLater events are just skipped by qt there.
I tried to simply replace your deleteLater calls by delete but, as expected, this causes crashes, because the widgets are often used after the delete calls. It would be really cool, if you could fix that sometime. As a workaround I currently simply replaced some deleteLater  calls by delete, wich solves the problem, but is more a hack then a final solution.

**Comment by githubuser0xFFFF** [comment]
In which weird case is step 3-6 called without returning to top event loop? This sounds completely wrong for me and I have to say sorry but I will not spend any time in fixing this. You need to fork this project.

**Comment by steffenroeber** [comment]
Imagine a menu action, that does exactly these steps at once. Then, event loop returns after last step but not in between. I guess that no very weird use case.

Gesendet von Yahoo Mail auf Android 
 
  Am Sa., Nov. 26, 2022 at 17:32 schrieb ***@***.***>:   


Closed #470 as completed.

**Comment by githubuser0xFFFF** [comment]
Weird :0)

**Comment by steffenroeber** [comment]
Let's call this menu action, open project. And imagine a project has widgets. Within a project someone creates widgets. open project now closes all widgets, save their states and opens and restores the other project widgets...😉

Gesendet von Yahoo Mail auf Android 
 
  Am Sa., Nov. 26, 2022 at 20:07 schrieb ***@***.***>:   


Weird :0)
